src/messaging/ExchangeMgr.cpp

```diff
@@ -81,7 +81,10 @@ CHIP_ERROR ExchangeManager::Init(NodeId localNodeId, TransportMgrBase * transpor
 
     for (auto & handler : UMHandlerPool)
     {
-        handler = {};
+        // Mark all handlers as unallocated.  This handles both initial
+        // initialization and the case when the consumer shuts us down and
+        // then re-initializes without removing registered handlers.
+        handler.Delegate = nullptr;
```

Comment:
A Reset() method may be better as it localizes knowledge about what a handler data contents is.

Reply:
https://github.com/project-chip/connectedhomeip/issues/5605